rest_framework/fields.py

```diff
@@ -576,10 +577,20 @@ def run_validation(self, data=empty):
         return super(CharField, self).run_validation(data)
 
     def to_internal_value(self, data):
-        return six.text_type(data)
+        value = six.text_type(data)
+
+        if self.trim_whitespace:
+            return value.strip()
+
+        return value
 
     def to_representation(self, value):
-        return six.text_type(value)
+        representation = six.text_type(value)
+
+        if self.trim_whitespace:
+            return representation.strip()
```

Comment:
@jpadilla Noted. In which case let's remove that behavior from `EmailField.to_representation`. Also after this change it looks like we don't need a custom `EmailField. to_internal_value` method any more either. Actually we should just drop both those methods entirely.

I think there's a point of clarity that `to_representation` should be the type coercion, as we can assume that normalization has already been applied. Passing every single string that goes through a REST framework serializer into `.strip()` would suck a little bit,